apis/Google.Cloud.Logging.Console/Google.Cloud.Logging.Console.Tests/GoogleCloudConsoleFormatterAugmentationTest.cs

```diff
@@ -0,0 +1,109 @@
+
+    private const string ExpectedAugmentedLogEntryJson = "{\"message\":\"test\",\"category\":\"LogCategory\",\"severity\":\"INFO\",\"simple_key\":\"simple_value\"}\n";
+    public class SimpleAugmenter : IGoogleCloudConsoleLogAugmenter
+    {
+        /// <summary>
+        /// Augment with a simple key-value pair.
+        /// </summary>
+        public void AugmentLog<T>(LogEntry<T> logEntry, Utf8JsonWriter writer)
+        {
+            writer.WriteString("simple_key", "simple_value");
+        }
+    }
+    private readonly GoogleCloudConsoleFormatterOptions simpleOptions = new GoogleCloudConsoleFormatterOptions { LogAugmenter = new SimpleAugmenter() };
+
+    [Fact]
+    public void SimplestLog()
+    {
+        var actualJson = LogSimpleLogEntry(simpleOptions);
+        Assert.Equal(ExpectedAugmentedLogEntryJson, actualJson);
+    }
+
+    private class ErrorAnnotationAugmenter : IGoogleCloudConsoleLogAugmenter
+    {
+        private static readonly JsonEncodedText s_jsonPayloadTypePropertyName = JsonEncodedText.Encode("@type");
+        private static readonly JsonEncodedText s_jsonPayloadTypePropertyValue = JsonEncodedText.Encode("type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent");
+        public void AugmentLog<T>(LogEntry<T> logEntry, Utf8JsonWriter writer)
+        {
+            if (logEntry.LogLevel >= LogLevel.Error)
+            {
+                writer.WriteString(s_jsonPayloadTypePropertyName, s_jsonPayloadTypePropertyValue);
+            }
+        }
+    }
+    private readonly GoogleCloudConsoleFormatterOptions errorAnnotationOptions = new GoogleCloudConsoleFormatterOptions { LogAugmenter = new ErrorAnnotationAugmenter() };
+
+    [Theory]
+    [InlineData(LogLevel.Trace)]
+    [InlineData(LogLevel.Debug)]
+    [InlineData(LogLevel.Information)]
+    [InlineData(LogLevel.Warning)]
+    public void NonErrorLogTest(LogLevel logLevel)
+    {
+        LogEntry<string> logEntry = new LogEntry<string>(logLevel, "LogCategory", new EventId(1), "Not so bad", exception: null, (state, exception) => state);
+        var actualJson = LogLogEntry(logEntry, errorAnnotationOptions);
+        Assert.DoesNotContain("@type", actualJson);
+        Assert.DoesNotContain("type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent", actualJson);
+    }
+
+    [Theory]
+    [InlineData(LogLevel.Error)]
+    [InlineData(LogLevel.Critical)]
+    public void ErrorLogTest(LogLevel logLevel)
+    {
+        LogEntry<string> logEntry = new LogEntry<string>(logLevel, "LogCategory", new EventId(1), "Not so bad", exception: null, (state, exception) => state);
+        var actualJson = LogLogEntry(logEntry, errorAnnotationOptions);
+        Assert.Contains("\"@type\":\"type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent\"", actualJson);
+    }
```

Comment:
All of this seems like something suitable for your codebase, but not for here. You are even testing the logic that augmentation happens only for a certain error level, that's very particular of log augmenter implementation that should be tested by the log augmenter author.

Maybe substitute by something that instead of logging a simple field logs a more complex JSON, but that's about as far as we need to go here.